services/src/test/java/org/exoplatform/onlyoffice/OnlyofficeEditorServiceTest.java

```diff
@@ -354,31 +553,248 @@ public void testSetLastModifier() throws Exception {
+    }
+  }
+
+  /**
+   * Test update Document for unknown user
+   */
+  @Test
+  public void testUpdateDocumentForUnknownUser() throws Exception {
     // Given
     startSessionAs("john");
     Node node = createDocument("Test Document.docx", "nt:file", "testContent", true);
+    Config config = editorService.createEditor("http", "127.0.0.1", 8080, "john", null, node.getUUID());
+    DocumentStatus status = new DocumentStatus.Builder().status(5L)
+                                                        .users(new String[] { "Thomas" })
+                                                        .userId("Thomas")
+                                                        .saved(true)
+                                                        .key(config.getDocument().getKey())
+                                                        .build();
+
+    // When
+    try {
+      editorService.updateDocument(status);
+      fail();
+    } catch (BadParameterException e) {
+      assertTrue(e instanceof BadParameterException);
+    } finally {
+      node.remove();
+    }
+  }
+
+  /**
+   * Test update Document for unknown file key
+   */
+  @Test
+  public void testUpdateDocumentForUnknownFileKey() throws Exception {
+    // Given
+    startSessionAs("john");
+    Node node = createDocument("Test Document.docx", "nt:file", "testContent", true);
+    DocumentStatus status = new DocumentStatus.Builder().status(5L)
+                                                        .users(new String[] { "john" })
+                                                        .userId("john")
+                                                        .saved(true)
+                                                        .key("key")
+                                                        .build();
+
+    // When
+    try {
+      editorService.updateDocument(status);
+      fail();
+    } catch (BadParameterException e) {
+      assertTrue(e instanceof BadParameterException);
+    } finally {
+      node.remove();
+    }
+  }
+
+  /**
+   * Test update Document for status code 6 Forcedsave done, save the version
+   * with its URL
+   */
+  @Test
+  public void testUpdateDocumentForceSaveDone() throws Exception {
+    // Given
+    startSessionAs("john");
+    Node node = createDocument("Test Document.docx", "nt:file", "testContent", true);
+    Config config = editorService.createEditor("http", "127.0.0.1", 8080, "john", null, node.getUUID());
+    DocumentStatus status = new DocumentStatus.Builder().status(6L)
+                                                        .users(new String[] { "john" })
+                                                        .userId("john")
+                                                        .saved(true)
+                                                        .key(config.getDocument().getKey())
+                                                        .build();
+
+    // When
+    editorService.updateDocument(status);
+
+    // Then
```

Comment:
Can we use editorService.getDocumentById to retrieve the document and make some assertions on it ?